src/main/java/org/javers/organization/structure/audit/AuditController.java

```diff
@@ -0,0 +1,115 @@
+        QueryBuilder jqlQuery = QueryBuilder.byClass(Person.class);
+
+        jqlQuery = param.isPresent() ? jqlQuery.andProperty(param.get()) : jqlQuery;
+
+        List<Change> changes = javers.findChanges(jqlQuery.build());
+
+        changes.sort((o1, o2) -> -1 * o1.getCommitMetadata().get().getCommitDate().compareTo(o2.getCommitMetadata().get().getCommitDate()));
+
+        JsonConverter jsonConverter = javers.getJsonConverter();
+
+        return jsonConverter.toJson(changes);
+    }
+
+    @RequestMapping("/person/{login}")
+    public String getPersonChanges(@PathVariable String login, @RequestParam Optional<String> param) {
+        QueryBuilder jqlQuery = QueryBuilder.byInstanceId(login, Person.class);
+
+        jqlQuery = param.isPresent() ? jqlQuery.andProperty(param.get()) : jqlQuery;
+
+        List<Change> changes = javers.findChanges(jqlQuery.build());
+
+        changes.sort((o1, o2) -> -1 * o1.getCommitMetadata().get().getCommitDate().compareTo(o2.getCommitMetadata().get().getCommitDate()));
+
+        JsonConverter jsonConverter = javers.getJsonConverter();
+
+        return jsonConverter.toJson(changes);
+    }
+
+    @RequestMapping("/person/snapshots")
+    public String getPersonSnapshots(@RequestParam Optional<String> param) {
+        QueryBuilder jqlQuery = QueryBuilder.byClass(Person.class);
+
+        jqlQuery = param.isPresent() ? jqlQuery.andProperty(param.get()) : jqlQuery;
+
+        List<CdoSnapshot> changes = javers.findSnapshots(jqlQuery.build());
+
+        changes.sort((o1, o2) -> -1 * o1.getCommitMetadata().getCommitDate().compareTo(o2.getCommitMetadata().getCommitDate()));
+
+        JsonConverter jsonConverter = javers.getJsonConverter();
+
+        return jsonConverter.toJson(changes);
+    }
+
+    @RequestMapping("/person/{login}/snapshots")
+    public String getPersonSnapshots(@PathVariable String login, @RequestParam Optional<String> param) {
+        QueryBuilder jqlQuery = QueryBuilder.byInstanceId(login, Person.class);
+
+        jqlQuery = param.isPresent() ? jqlQuery.andProperty(param.get()) : jqlQuery;
+
+        List<CdoSnapshot> changes = javers.findSnapshots(jqlQuery.build());
+
+        changes.sort((o1, o2) -> -1 * o1.getCommitMetadata().getCommitDate().compareTo(o2.getCommitMetadata().getCommitDate()));
+
+        JsonConverter jsonConverter = javers.getJsonConverter();
+
+        return jsonConverter.toJson(changes);
+    }
+
+    @RequestMapping("/hierarchy/{left}/diff/{right}")
+    public String getPersonSnapshots(@PathVariable String left, @PathVariable String right) {
+        Hierarchy l = hierarchyService.find(left);
+        Hierarchy p = hierarchyService.find(right);
+
+
+        Diff diff = javers.compare(l, p);
+
+
+//        TODO
+//        List<Change> changes = diff.getChanges(input ->
+//                (input instanceof NewObject
+//                        && input.getAffectedGlobalId().getCdoClass().getClientsClass() != Hierarchy.class));
```

Comment:
Tutaj chciałem filrować po danej klasie ale api się coś zmieniło i ten kod się nie kompiluje teraz, wymaga to jeszcze chwili pracy.